### PR TITLE
[auth-swift] Custom Auth Domains

### PR DESCRIPTION
**Pull request by paulb777**
Swift implementation of #11619 

#no-changelog

### CODE REVIEW

File: FirebaseAuth/Sources/Swift/Utilities/AuthWebUtils.swift [hunk 1]
```diff
@@ -77,6 +77,29 @@ import Foundation
...
+
+    // Check for the presence of a scheme (e.g., http:// or https://)
+    if domain.prefix(7).caseInsensitiveCompare("http://") == .orderedSame {
+      domain = String(domain.dropFirst(7))
+    } else if domain.prefix(8).caseInsensitiveCompare("https://") == .orderedSame {
+      domain = String(domain.dropFirst(8))
+    }
+
+    // Remove trailing slashes
+    domain = (domain as NSString).standardizingPath as String
```
Comment (ncooke3) [inline FirebaseAuth/Sources/Swift/Utilities/AuthWebUtils.swift:95]: nit: any reason not to do this on line 85?

**Review by ncooke3 — APPROVED** [review]
nice!

File: FirebaseAuth/Sources/Swift/Utilities/AuthWebUtils.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseAuth/Sources/Swift/Utilities/AuthWebUtils.swift:95]: Yes. That was my first implementation, but it consolidates the "//" after `http:` to a single slash and would make the code less readable.

Comment (ncooke3) [reply FirebaseAuth/Sources/Swift/Utilities/AuthWebUtils.swift:95]: ah, makes sense. Thanks for the explanation.

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExampleUITests/AuthenticationExampleUITests.swift [hunk 2]
```diff
@@ -38,7 +38,7 @@ class AuthenticationExampleUITests: XCTestCase {
 
   func testAuthOptions() {
     // There are 13 sign in methods, each with its own cell
```
Comment (pragatimodi) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExampleUITests/AuthenticationExampleUITests.swift:40]:
```suggestion
    // There are 15 sign in methods, each with its own cell
```

**Review by pragatimodi — APPROVED** [review]

**Pull request merged**